snapshots.go

```diff
@@ -0,0 +1,136 @@
+package godo
+
+import "fmt"
+
+const snapshotBasePath = "v2/snapshots"
+
+// SnapshotsService is an interface for interfacing with the snapshots
+// endpoints of the DigitalOcean API
+// See: https://developers.digitalocean.com/documentation/v2#snapshots
+type SnapshotsService interface {
+	List(*ListOptions) ([]Snapshot, *Response, error)
+	ListVolume(*ListOptions) ([]Snapshot, *Response, error)
+	ListDroplet(*ListOptions) ([]Snapshot, *Response, error)
+	Get(string) (*Snapshot, *Response, error)
+	Delete(string) (*Response, error)
+}
+
+// SnapshotsServiceOp handles communication with the snapshot related methods of the
+// DigitalOcean API.
+type SnapshotsServiceOp struct {
+	client *Client
+}
+
+var _ SnapshotsService = &SnapshotsServiceOp{}
+
+// Snapshot represents a DigitalOcean Snapshot
+type Snapshot struct {
+	ID            string   `json:"id,omitempty"`
+	Name          string   `json:"name,omitempty"`
+	ResourceID    string   `json:"type,omitempty"`
+	ResourceType  string   `json:"type,omitempty"`
+	Regions       []string `json:"regions,omitempty"`
+	MinDiskSize   int      `json:"min_disk_size,omitempty"`
+	SizeGigabytes int      `json:"size_gigabytes,omitempty"`
```

Comment:
the rest of the code uses `SizeGigaBytes` (capitalized `B`):
* https://github.com/digitalocean/godo/blob/master/storage.go#L79
* https://github.com/digitalocean/godo/blob/master/storage.go#L53